llama-index-core/llama_index/core/schema.py

```diff
@@ -960,10 +960,13 @@ def __init__(self, **data: Any) -> None:
         super().__init__(**data)
 
     @model_serializer(mode="wrap")
-    def custom_model_dump(self, handler: Any) -> Dict[str, Any]:
+    def custom_model_dump(
+        self, handler: SerializerFunctionWrapHandler, info: SerializationInfo
+    ) -> Dict[str, Any]:
         """For full backward compatibility with the text field, we customize the model serializer."""
         data = super().custom_model_dump(handler)
-        data["text"] = self.text
+        if "text" not in info.exclude:
```

Comment:
should we support any field to exclude?

Reply:
we do already for the other fields.`super().custom_model_dump(handler, info)` calls the default serialization method under the hood, which excludes everything else we pass to exclude. We needed to add this for `text` because we added that to `data` after the "exclude fields" logic was executed.
In the test I added some more fields to exclude to confirm this is the case

Reply:
gotcha!